src/Database/Redis/Connection.hs

```diff
@@ -282,20 +294,21 @@ refreshShardMapWithNodeConn :: [Cluster.NodeConnection] -> IO ShardMap
 refreshShardMapWithNodeConn [] = throwIO $ ClusterConnectError (Error "Couldn't refresh shardMap due to connection error")
 refreshShardMapWithNodeConn nodeConnsList = do
     selectedIdx <- randomRIO (0, (length nodeConnsList) - 1)
-    let (Cluster.NodeConnection ctx _ _) = nodeConnsList !! selectedIdx
-    pipelineConn <- PP.fromCtx ctx
-    envTimeout <- fromMaybe (10 ^ (3 :: Int)) . (>>= readMaybe) <$> lookupEnv "REDIS_CLUSTER_SLOTS_TIMEOUT"
-    raceResult <- race (threadDelay envTimeout) (try $ refreshShardMapWithConn pipelineConn True) -- racing with delay of default 1 ms 
-    case raceResult of
-        Left () -> do
-            print $ "TimeoutForConnection " <> show ctx 
-            throwIO $ Cluster.TimeoutException "ClusterSlots Timeout"
-        Right eiShardMapResp -> 
-            case eiShardMapResp of
-                Right shardMap -> pure shardMap 
-                Left (err :: SomeException) -> do
-                    print $ "ShardMapRefreshError-" <> show err 
-                    throwIO $ ClusterConnectError (Error "Couldn't refresh shardMap due to connection error")
+    let (Cluster.NodeConnection pool _ _) = nodeConnsList !! selectedIdx
+    withResource pool $ \ctx -> do
+        pipelineConn <- PP.fromCtx ctx
+        envTimeout <- fromMaybe (10 ^ (3 :: Int)) . (>>= readMaybe) <$> lookupEnv "REDIS_CLUSTER_SLOTS_TIMEOUT"
```

Comment:
Same comment, lets combine all such configs in a single place to be more readable / trackable.